src/iceberg/test/rest_catalog_test.cc

```diff
@@ -19,103 +19,206 @@
+
+constexpr std::string_view kDockerProjectName = "iceberg-rest-catalog-service";
+constexpr std::string_view kCatalogName = "test_catalog";
+constexpr std::string_view kWarehouseName = "default";
+constexpr std::string_view kLocalhostUri = "http://localhost";
+
+/// \brief Check if a localhost port is ready to accept connections
+/// \param port Port number to check
+/// \return true if the port is accessible on localhost, false otherwise
+bool CheckServiceReady(uint16_t port) {
+  int sock = socket(AF_INET, SOCK_STREAM, 0);
+  if (sock < 0) {
+    return false;
+  }
+
+  struct timeval timeout{
+      .tv_sec = 1,
+      .tv_usec = 0,
+  };
+  setsockopt(sock, SOL_SOCKET, SO_RCVTIMEO, &timeout, sizeof(timeout));
+
+  sockaddr_in addr{
+      .sin_family = AF_INET,
+      .sin_port = htons(port),
+      .sin_addr = {.s_addr = htonl(INADDR_LOOPBACK)}  // 127.0.0.1
+  };
+  bool result =
+      (connect(sock, reinterpret_cast<struct sockaddr*>(&addr), sizeof(addr)) == 0);
+  close(sock);
+  return result;
+}
+
+}  // namespace
+
+/// \brief Integration test fixture for REST catalog with automatic Docker Compose setup。
+class RestCatalogIntegrationTest : public ::testing::Test {
  protected:
-  void SetUp() override {
-    // Default configuration for local testing
-    // You can override this with environment variables if needed
-    const char* uri_env = std::getenv("ICEBERG_REST_URI");
-    const char* warehouse_env = std::getenv("ICEBERG_REST_WAREHOUSE");
+  static void SetUpTestSuite() {
+    std::string project_name{kDockerProjectName};
+    std::filesystem::path resources_dir =
+        std::filesystem::path(__FILE__).parent_path() / "resources";
+
+    // Create and start DockerCompose
+    docker_compose_ = std::make_unique<DockerCompose>(project_name, resources_dir);
+    docker_compose_->Up();
+
+    // Wait for REST catalog to be ready on localhost
+    std::println("[INFO] Waiting for REST catalog to be ready at localhost:{}...",
+                 kRestCatalogPort);
+    for (int i = 0; i < kMaxRetries; ++i) {
+      if (CheckServiceReady(kRestCatalogPort)) {
+        std::println("[INFO] REST catalog is ready!");
+        return;
+      }
+      std::println(
+          "[INFO] Waiting for 1s for REST catalog to be ready... (attempt {}/{})", i + 1,
+          kMaxRetries);
+      std::this_thread::sleep_for(std::chrono::milliseconds(kRetryDelayMs));
+    }
+    throw RestError("REST catalog failed to start within {} seconds", kMaxRetries);
+  }
 
-    std::string uri = uri_env ? uri_env : "http://localhost:8181";
-    std::string warehouse = warehouse_env ? warehouse_env : "default";
+  static void TearDownTestSuite() { docker_compose_.reset(); }
 
+  void SetUp() override {
     config_ = RestCatalogProperties::default_properties();
-    config_->Set(RestCatalogProperties::kUri, uri)
-        .Set(RestCatalogProperties::kName, std::string("test_catalog"))
-        .Set(RestCatalogProperties::kWarehouse, warehouse);
+    config_
```

Comment:
It seems that we don't need to keep the config instance since it is a one-off thing. You may directly create a config in the `CreateCatalog()` and throw it away after creating the catalog.